bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/type/UidUtils.java

```diff
@@ -0,0 +1,217 @@
+            for (int idx = 0; idx < 4; idx++) {
+                result = result.replace("CC" + idx, "CC");
+                result = result.replace("HC" + idx, "HC");
+            }
+        }
+
+        result = result.toLowerCase();
+
+        if (result.indexOf("history") >= 0) {
+            for (int idx = 0; idx < 20; idx++) {
+                result = result.replace("history " + idx, "history");
+            }
+        }
+
+        result = result.replace(" mon", "");
+        result = result.replace(" yue", "");
+        result = result.replace(" wed", "");
+        result = result.replace(" thu", "");
+        result = result.replace(" tue", "");
+        result = result.replace(" fri", "");
+        result = result.replace(" sat", "");
+        result = result.replace(" sun", "");
+        result = result.replace(" mo", "");
+        result = result.replace(" tu", "");
+        result = result.replace(" we", "");
+        result = result.replace(" th", "");
+        result = result.replace(" fr", "");
+        result = result.replace(" sa", "");
+        result = result.replace(" su", "");
+
+        if (result.indexOf("holidays") >= 0) {
+            if (result.indexOf("firstd") >= 0) {
+                result = "holidays-hc-firstd";
+            }
+            if (result.indexOf("lastd") >= 0) {
+                result = "holidays-hc-lastd";
+            }
+        }
+
+        result = result.replace("---", "-");
+        result = result.replace("--", "-");
+        result = result.replace('\'', '-');
+        result = result.replace('/', '-');
+        result = result.replace(' ', '-');
+        result = result.replace("+", "-");
+        result = result.replace("--", "-");
+
+        result = result.replace("standard-tsp-hc", "time-switch-program-standard");
+        result = result.replace("standard-tsp-4", "time-switch-program-standard");
+        result = result.replace("tsp-3", "time-switch-program-day");
+        result = result.replace("tsp-4", "time-switch-program-day");
+        result = result.replace("setpointtemp", "setpoint-temp-");
+        result = result.replace("rmtmp", "roomtemp");
+        result = result.replace("roomtempfrostprot", "room-temp-frostprot-");
+        result = result.replace("-setp", "-setpoint");
+        result = result.replace("optg", "operating-");
+        result = result.replace("-comf", "-comfort");
+        result = result.replace("-red", "-reduce");
+        result = result.replace("setp-", "-setpoint");
+        result = result.replace("roomtemp-", "room-temp-");
+        result = result.replace("-setpointhc", "-setpoint-hc");
+        result = result.replace("setphc", "-setpoint-hc");
+
+        return result;
+    }
+
+    /**
+     * Generates the ChannelTypeUID for the given datapoint with deviceType, channelNumber and datapointName.
+     */
+    public static ChannelTypeUID generateChannelTypeUID(SiemensHvacMetadataDataPoint dpt) throws Exception {
```

Comment:
Consider adding unit test coverage for this method, especially to document what `normalizeDescriptor` does.

Reply:
Just to be clear: Although I'm a big fan of unit tests, with "consider" I mean this comment is optional and up to you.